### ISSUE TITLE
Constructor as a sink method

### ISSUE THREAD

**Issue by dencat** [gmu-swe/phosphor#103]
Hello!
I'm trying to track the transfer of tainted data to the constructor, but it does not work. For example:
```
package com.mycompany.app;

import java.io.File;

public class App {

    public App() {}

    public static void main(String[] args) {
        App a = new App();
        String z = a.source();

   	// ok
        //a.sink(z);

	// doesn't work
        File file = new File(z);
    }

    public String source() {
      return "test";
    }

    public void sink(String arg) {

    }
}
```
source.txt:
```
com/mycompany/app/App.source()Ljava/lang/String;
```
sink.txt:
```
com/mycompany/app/App.sink(Ljava/lang/String;)V
java/io/File.<init>(Ljava/lang/String;)V
```
JAVA was instrumented with controlTrack argument:
```
java -jar Phosphor-0.0.4-SNAPSHOT.jar -controlTrack $JAVA_HOME java_inst
```

Thank you!

**Comment by jon-bell** [comment]
Hi,
This is now fixed, with tests in 7efc3e06c26ecc735e2a38b133a9ab4cc6e703df

If you run into more issues, please create minimized test cases that show the failing behavior (you can see examples of tests that reveal this behavior in the recent commits) - that makes our job of reproducing and fixing the issue much easier.

**Comment by dencat** [comment]
Thank you.
Awesome project
